controllers/perconaservermysqlbackup_controller.go

```diff
@@ -45,16 +68,229 @@ type PerconaServerMySQLBackupReconciler struct {
+	}
+
+	if cluster.Status.MySQL.State != apiv1alpha1.StateReady {
+		l.Info("Cluster is not ready", "cluster", cr.Name)
+		return rr, nil
+	}
+
+	j := &batchv1.Job{}
+	nn = xtrabackup.NamespacedName(cr)
+	err := r.Client.Get(ctx, nn, j)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return rr, errors.Wrapf(err, "get job %v", nn.String())
+	}
+
+	if k8serrors.IsNotFound(err) {
+		l.Info("Creating backup job", "jobName", nn.Name)
+
+		initImage, err := k8s.InitImage(ctx, r.Client)
+		if err != nil {
+			return rr, errors.Wrap(err, "get operator image")
+		}
+
+		destination := fmt.Sprintf("%s-%s-full", cr.Spec.ClusterName, cr.CreationTimestamp.Format("2006-01-02-15:04:05"))
+		job := xtrabackup.Job(cluster, cr, destination, initImage, storage)
+
+		switch storage.Type {
+		case apiv1alpha1.BackupStorageS3:
+			if storage.S3 == nil {
+				return rr, errors.New("s3 stanza is required in storage")
+			}
+
+			nn := types.NamespacedName{Name: storage.S3.CredentialsSecret, Namespace: cr.Namespace}
+			exists, err := k8s.ObjectExists(ctx, r.Client, nn, &corev1.Secret{})
+			if err != nil {
+				return rr, errors.Wrapf(err, "check %s exists", nn)
+			}
 
-	// your logic here
+			if !exists {
+				return rr, errors.Errorf("secret %s not found", nn)
+			}
 
-	return ctrl.Result{}, nil
+			if err := xtrabackup.SetStorageS3(job, storage.S3); err != nil {
+				return rr, errors.Wrap(err, "set storage S3")
+			}
+
+			status.Destination = fmt.Sprintf("s3://%s/%s", storage.S3.Bucket, destination)
+		case apiv1alpha1.BackupStorageGCS:
+			if storage.GCS == nil {
+				return rr, errors.New("gcs stanza is required in storage")
+			}
+
+			nn := types.NamespacedName{Name: storage.GCS.CredentialsSecret, Namespace: cr.Namespace}
+			exists, err := k8s.ObjectExists(ctx, r.Client, nn, &corev1.Secret{})
+			if err != nil {
+				return rr, errors.Wrapf(err, "check %s exists", nn)
+			}
+
+			if !exists {
+				return rr, errors.Errorf("secret %s not found", nn)
+			}
+
+			if err := xtrabackup.SetStorageGCS(job, storage.GCS); err != nil {
+				return rr, errors.Wrap(err, "set storage GCS")
+			}
+
+			status.Destination = fmt.Sprintf("gcs://%s/%s", storage.GCS.Bucket, destination)
+		case apiv1alpha1.BackupStorageAzure:
+			if storage.Azure == nil {
+				return rr, errors.New("azure stanza is required in storage")
+			}
+
+			nn := types.NamespacedName{Name: storage.Azure.CredentialsSecret, Namespace: cr.Namespace}
+			exists, err := k8s.ObjectExists(ctx, r.Client, nn, &corev1.Secret{})
+			if err != nil {
+				return rr, errors.Wrapf(err, "check %s exists", nn)
+			}
+
+			if !exists {
+				return rr, errors.Errorf("secret %s not found", nn)
+			}
+
+			if err := xtrabackup.SetStorageAzure(job, storage.Azure); err != nil {
+				return rr, errors.Wrap(err, "set storage Azure")
+			}
+
+			status.Destination = fmt.Sprintf("%s/%s", storage.Azure.ContainerName, destination)
```

Comment:
```suggestion
			status.Destination = fmt.Sprintf("azure://%s/%s", storage.Azure.ContainerName, destination)
```

Reply:
The idea of having `s3://` and `gcs://` in the destination that they're usable with `aws` and `gsutil` but Azure uses an URI like `https://myaccount.blob.core.windows.net/mycontainer/myblob`. IMO it's too long to put